lib/surface/compiler/eex_engine.ex

```diff
@@ -498,44 +498,31 @@ defmodule Surface.Compiler.EExEngine do
   end
 
   # Function component
-  defp build_slot_props(%AST.FunctionComponent{fun: fun} = component, buffer, state, _context_var)
-       when fun != nil do
-    slot_info =
-      component.slot_entries
-      |> Enum.map(fn {name, slot_entries} ->
-        state = %{state | scope: [:slot_entry | state.scope]}
+  defp build_slot_props(%AST.FunctionComponent{fun: fun} = component, buffer, state, _ctx_var) when fun != nil do
+    for {name, slot_entries} <- component.slot_entries,
+        state = %{state | scope: [:slot_entry | state.scope]},
+        nested_slot_entries = handle_slot_entries(component, slot_entries, buffer, state),
+        not Enum.empty?(nested_slot_entries) do
```

Comment:
I have a question here, instead of using `Enum.empty?(nested_slot_entries)` can we check `nested_slot_entries != []`? If it is possible can we change? WDYT?